tools/schemacode/schemacode/tests/test_schema.py

```diff
@@ -27,3 +27,109 @@ def test_object_definitions(schema_obj):
 
             assert "name" in obj_def.keys(), obj_key
             assert "description" in obj_def.keys(), obj_key
+
+
+def test_formats(schema_obj):
+    """Test valid string patterns allowed by the specification."""
+    import re
+
+    # Check that valid strings match the search pattern.
+    GOOD_PATTERNS = {
+        "label": ["01", "test", "test01", "Test01"],
+        "index": ["01", "1", "10000", "00001"],
+        "string": ["any string is valid."],
+        "integer": ["5", "10", "-5", "-10"],
+        "number": [
+            "5",  # integers are allowed
+            "3.14",  # floats too
+            "-5",  # they can be negative
+            "-3.14",
+            "1e3",  # scientific notation is allowed
+            "-2.1E+5",
+        ],
+        "boolean": ["true", "false"],
+        "date": ["2022-01-05", "2022-01-05UTC", "2022-50-50"],
+        "datetime": [
+            "2022-01-05T13:16:30",
+            "2022-01-05T13:16:30.5",  # subsecond resolution is allowed
+            "2022-01-05T13:16:30.000005",  # up to 6 decimal points
+            "2022-01-05T13:16:30UTC",  # timezones are allowed
+            "2022-01-05T13:16:30.05UTC",
+        ],
+        "time": [
+            "13:16:30",
+            "09:00:00",
+            "9:00:00",  # leading zeros are not required for hours
+        ],
+        "unit": ["any string is valid."],
```

Comment:
oh, even with spaces and unicodes?  I thought we had discussion and there was a decision on some unicodes to be ok (but not any), then with #411 CMIXF-12 ascii recommended. https://github.com/bids-standard/bids-specification/blob/HEAD/src/99-appendices/05-units.md mentions `µ` (could be only as prefix) `Ω` (only as suffix) and `°` (all by itself?).  So may be regex and test here could be made more stringent?  well, aforementioned page gives example just as "NOT RECOMMENDED" instead of "WRONG": `volts per microsecond` so may be my whining is all in vein?  But then I would restrict to composition of alphanumerics, space, `-` and those 3 unicodes, and thus disallow everything else.

Reply:
TBH I allowed any string because I don't know much about the unit validation. Do you mind if we allow any string in this PR and make sure to follow up with a more accurate regex before the next release? Assuming we merge this after 1.7.0 is released, so there'll be a bit of time between 1.7.0 and 1.8.0.

Reply:
I would not mind. Please just create an issue for that so we do not forget?

Reply:
Done. #999.